src/lambda-wrapper.ts

```diff
@@ -29,10 +35,16 @@ const ApiGatewayHandler = (router: Middleware, onFinished: OnFinishedHandler) =>
    * @param {object} context API Gateway context object
    * @return {promise} Returns undefined if callback param is set. Return a promise if callback param is undefined.
    */
-  const handleApiGatewayEvent: APIGatewayProxyHandler = function (event, context) {
-    return new Promise(resolve => {
-      const req = new Request(event)
-      const res = (req.res = new Response(req, async (err, out) => {
+  const handleApiGatewayEvent: APIGatewayProxyHandler | APIGatewayProxyHandlerV2 = function (
+    event: (APIGatewayProxyEvent & { version: string }) | APIGatewayProxyEventV2,
+    context: any
+  ) {
+    return new Promise<APIGatewayProxyResult>(resolve => {
+      const req =
+        event.version == '2.0'
```

Comment:
maybe here something like `if ('version' in event && event.version === '2.0')` would simplify things and you could remove the `event as ...`